File: galgamecommon/src/main/java/com/goexp/galgame/common/util/Network.java
package com.goexp.galgame.common.util;


import com.goexp.galgame.common.Config;

import java.util.ResourceBundle;

public class Network {
    private static boolean isInit = false;

    public static void initProxy() {

        if (Config.proxy && !isInit) {

            var prop = ResourceBundle.getBundle("proxy");
            prop.keySet().forEach(key -> {
                System.setProperty(key, prop.getString(key));
            });

            isInit = true;
        }
    }
}

File: galgamecommon/src/main/scala/com/goexp/galgame/common/util/Network.scala
package com.goexp.galgame.common.util

import java.util.ResourceBundle

import com.goexp.galgame.common.Config
import com.typesafe.scalalogging.Logger

object Network {

  protected val logger: Logger = Logger(Network.getClass)

  private var isInit = false

  def initProxy(): Unit =
    if (Config.proxy && !isInit) {

      logger.info("init Network")

      val prop = ResourceBundle.getBundle("proxy")


      prop.keySet.forEach(k => {
        val v = prop.getString(k)

        logger.debug("{}->{}",k,v)

        System.setProperty(k, v)
      })


      isInit = true
    }
}